.migration/docs/setup-guide.md

```diff
@@ -0,0 +1,204 @@
+
+```bash
+cp path/to/s2dm/.migration/workflows/migrate.yml .github/workflows/migrate.yml
+cp path/to/s2dm/.migration/.bumpversion.toml .bumpversion.toml
+```
+
+#### Option D: Using curl
+
+```bash
+mkdir -p .github/workflows
+
+curl -o .github/workflows/migrate.yml https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/workflows/migrate.yml
+
+curl -o .bumpversion.toml https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/.bumpversion.toml
+```
+
+### 2. Configure Repository Variables
+
+The workflow supports optional configuration through GitHub repository variables.
+
+Go to your repository → Settings → Secrets and variables → Actions → Variables tab, and add:
+
+#### SKOS Generation Variables
+
+- **`SKOS_NAMESPACE`**: The namespace URI for SKOS concepts (e.g., `https://example.org/vss#`)
+- **`SKOS_PREFIX`**: The prefix for SKOS concept URIs (e.g., `ns`)
+- **`SKOS_LANGUAGE`**: BCP 47 language tag for SKOS labels (default: `en`)
+
+#### Registry Variables
+
+- **`CONCEPT_NAMESPACE`**: The namespace URI for concept URIs (e.g., `https://example.org/vss#`)
+- **`CONCEPT_PREFIX`**: The prefix for concept URIs (e.g., `ns`)
+
+#### SHACL Generation Variables
+
+- **`SHACL_SERIALIZATION_FORMAT`**: RDF serialization format (default: `ttl`)
+- **`SHACL_SHAPES_NAMESPACE`**: Namespace for SHACL shapes (e.g., `http://example.ns/shapes#`)
+- **`SHACL_SHAPES_PREFIX`**: Prefix for SHACL shapes (e.g., `shapes`)
+- **`SHACL_MODEL_NAMESPACE`**: Namespace for data model (e.g., `https://example.ns/model#`)
+- **`SHACL_MODEL_PREFIX`**: Prefix for data model (e.g., `model`)
+
+> Note: All variables are optional. If not set, the commands will use their default values.
+
+### 3. Repository Structure
+
+Ensure your repository has the following structure:
+
+```plain
+your-repo/
+├── .github/
+│   └── workflows/
+│       └── migrate.yml          # Copied from COVESA/s2dm
+├── .bumpversion.toml            # Copied from COVESA/s2dm
+├── spec/                        # Your specification files (GraphQL)
+│   └── ...
+├── units.yaml                   # Units file
+└── README.md
+```
+
+## How the Workflow Works
+
+The migration workflow is triggered on pushes to the `main` branch when files in the `spec/` directory are modified.
+
+### Critical Workflow Steps
+
+#### 1. Version Bump Detection
+
+```yaml
+- name: Check spec version bump
+```
+
+- Compares current spec with the previous release
+- Determines if changes require a `major`, `minor`, `patch`, or no version bump
+- Skips the entire workflow if no version bump is needed
+
+#### 2. Schema Composition
+
+```yaml
+- name: Compose GraphQL schema
+```
+
+- Combines all specification files from `spec/` into a single GraphQL schema
+- Output: `.artifacts/graphql/schema.graphql`
+
+#### 3. JSON Schema Generation
+
+```yaml
+- name: Generate JSON schema
+```
+
+- Converts the GraphQL schema to JSON Schema format
+- Output: `.artifacts/jsonschema/schema.json`
```

Comment:
To avoid confusion to the reader, use consistent terminology. Instead of using similar terms like generate, convert, traslate, etc. Stick to the one that is aligned to the command group. i.e., `export`